src/ape/types/__init__.py

```diff
@@ -199,6 +199,34 @@ class ContractLog(BaseInterfaceModel):
     event_arguments: Dict[str, Any]
     """The arguments to the event, including both indexed and non-indexed data."""
 
+    @validator("contract_address", pre=True)
+    def validate_address(cls, value):
+        from ape import convert
+
+        return convert(value, AddressType)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, BaseContractLog):
+            return NotImplemented
```

Comment:
Add a note of how this triggers `x.__eq__(y)` to change to `y.__eq__(x)`, and how this fits in with `MockContractLog`